interview-questions.md

```diff
@@ -6,44 +6,72 @@ First, without external resources. Challenge yourself to answer from memory.
+
+
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: If a number has no decimal, in the Ruby language it is considered to be an integer. If a number has a decimal, it is considered to be a float.
 
 Researched answer:
+There are five number types in Ruby. There are Integer, Float, Complex, Rational, and Big Decimal. The difference been an Integer and a Float has to deal with whether a number is a whole number or an imperfect or impercise number like a decimal. Whole numbers are integers, while decimal numbers are considered Floats. In Ruby, the sum of two decimal numbers are considered false. While the sum of two integers are considered true.
+
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: In other languages like JavaScript, every function needs a return or else it won't print anything to the screen or display for the end-user. However, in Ruby returns are implicit in that if a method is being executed, the Ruby method already implies it is there. The return method is also optional. You can put one if you like but it is against the Ruby language convention.
 
 Researched answer:
+Whether or not you provide a return statement in a Ruby block of code, the interpreter will imply you meant to because the Ruby language has implicit returns. A block in Ruby implies that there is a return statement in there. In other languages like JavaScript, an explicit return is required for the function to execute and get the desired outcome, otherwise an undefined will be returned.
+
 
 4. What is a block in Ruby?
 
 Your answer:
+A block in Ruby is a chunk of code where you can execute some sort of action. Typically, you use a block of code with a method. A Ruby block is characterized by its 'def' and 'end' statements, anagolous to JavaScript's opening and closing curly braces.
 
 Researched answer:
+A Ruby block is a piece or block of code that is an anonymous function that gets passed to methods. Such a function can get passed into a method and is characterized by the 'do' and 'end' keywords. While in JavaScript curly braces are used. It should be noted that you can use curly braces in blocks, typically Rubyists will only use curly brackets in single-line statments per Ruby's styling convention. Also, a Ruby Block is similar to a JavaScript Closure.
+
 
-1. How do you extract a value in a Ruby hash?
+5. How do you extract a value in a Ruby hash?
 
 Your answer:
+In order to extract a value in a hash in Ruby you need to reference the hash name and then in brackets with a colon the Ruby key/symbol. For example, if there was a person hash with a key name and a value of "philip", you could get the value with 
+p person[:name]
+// Output: "philip"
 
 Researched answer:
+In Ruby, in order to get the value from a hash you use bracket notation. If you only want to extract one value from a hash you call on the hash and the desired key:
```

Comment:
Excellent work in this section!  Your phrasing in these answers is very well articulated and professional!